### DISCUSSION TITLE
0x1::timestamp doesnt work on tests??

### DISCUSSION THREAD

**Discussion by ajaythxkur** [aptos-labs/aptos-developer-discussions#376] [Questions] [answered]
`public fun now_microseconds(): u64 acquires CurrentTimeMicroseconds {
│    │                ---------------- In this function in 0x1::timestamp
│ 62 │         borrow_global<CurrentTimeMicroseconds>(@aptos_framework).microseconds
│    │         ^^^^^^^^^^^^^ Test was not expected to error, but it gave a MISSING_DATA (code 4008) error with error message: "Failed to borrow global resource from 0000000000000000000000000000000000000000000000000000000000000001". Error originating in the module 0000000000000000000000000000000000000000000000000000000000000001::timestamp rooted here`

timestamps dont work on tests??

**Comment by SukiratWarring** [comment, marked as answer]
You need to start the timestamp to setup the environment.

`timestamp::set_time_has_started_for_testing(aptos_framework);`
